tensorflow/contrib/eager/python/examples/generative_examples/dcgan.ipynb

```diff
@@ -1,324 +1,405 @@
-        "* **Discriminator**\n",
-        "  * **The discriminator is responsible for classifying the fake images from the real images.**\n",
-        "  * In other words, the discriminator is given generated images (from the generator) and the real MNIST images. The job of the discriminator is to classify these images into fake (generated) and real (MNIST images).\n",
-        "  * **Basically the generator should be good enough to fool the discriminator that the generated images are real**."
+        "We will use tf.keras [Sequential API](https://www.tensorflow.org/guide/keras#sequential_model) to define the generator and discriminator models."
       ]
     },
     {
-      "cell_type": "code",
-      "execution_count": 0,
       "metadata": {
-        "colab": {},
+        "colab_type": "text",
+        "id": "-tEyxE-GMC48"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "### The Generator Model\n",
+        "\n",
+        "The generator is responsible for creating convincing images that are good enough to fool the discriminator. The network architecture for the generator consists of [Conv2DTranspose](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Conv2DTranspose) (Upsampling) layers. We start with a fully connected layer and upsample the image two times in order to reach the desired image size of 28x28x1. We increase the width and height, and reduce the depth as we move through the layers in the network. We use [Leaky ReLU](https://www.tensorflow.org/api_docs/python/tf/keras/layers/LeakyReLU) activation for each layer except for the last one where we use a tanh activation."
+      ]
+    },
+    {
+      "metadata": {
+        "id": "6bpTcDqoLWjY",
         "colab_type": "code",
-        "id": "VGLbvBEmjK0a"
+        "colab": {}
       },
-      "outputs": [],
+      "cell_type": "code",
       "source": [
-        "class Generator(tf.keras.Model):\n",
-        "  def __init__(self):\n",
-        "    super(Generator, self).__init__()\n",
-        "    self.fc1 = tf.keras.layers.Dense(7*7*64, use_bias=False)\n",
-        "    self.batchnorm1 = tf.keras.layers.BatchNormalization()\n",
-        "    \n",
-        "    self.conv1 = tf.keras.layers.Conv2DTranspose(64, (5, 5), strides=(1, 1), padding='same', use_bias=False)\n",
-        "    self.batchnorm2 = tf.keras.layers.BatchNormalization()\n",
-        "    \n",
-        "    self.conv2 = tf.keras.layers.Conv2DTranspose(32, (5, 5), strides=(2, 2), padding='same', use_bias=False)\n",
-        "    self.batchnorm3 = tf.keras.layers.BatchNormalization()\n",
+        "def make_generator_model():\n",
+        "    model = tf.keras.Sequential()\n",
+        "    model.add(tf.keras.layers.Dense(7*7*256, use_bias=False, input_shape=(100,)))\n",
+        "    model.add(tf.keras.layers.BatchNormalization())\n",
+        "    model.add(tf.keras.layers.LeakyReLU())\n",
+        "      \n",
+        "    model.add(tf.keras.layers.Reshape((7, 7, 256)))\n",
```

Comment:
Optional: Maybe a comment here mentioning that `None` is the batch dimension?

Reply:
Done